libcnb-package/src/output.rs

```diff
@@ -0,0 +1,261 @@
+            e,
+        )
+    })?;
+
+    let destination_path = destination_path.as_ref();
+
+    fs::copy(
+        buildpack_path.as_ref(),
+        destination_path.join("buildpack.toml"),
+    )
+    .map_err(|e| {
+        AssembleBuildpackDirectoryError::WriteBuildpack(destination_path.join("buildpack.toml"), e)
+    })?;
+
+    let default_buildpackage = Buildpackage::default();
+    let buildpackage_content = toml::to_string(buildpackage.unwrap_or(&default_buildpackage))
+        .map_err(AssembleBuildpackDirectoryError::SerializeBuildpackage)?;
+
+    fs::write(destination_path.join("package.toml"), buildpackage_content).map_err(|e| {
+        AssembleBuildpackDirectoryError::WriteBuildpackage(destination_path.join("package.toml"), e)
+    })?;
+
+    let bin_path = destination_path.join("bin");
+    fs::create_dir_all(&bin_path)
+        .map_err(|e| AssembleBuildpackDirectoryError::CreateBinDirectory(bin_path.clone(), e))?;
+
+    fs::copy(
+        &buildpack_binaries.buildpack_target_binary_path,
+        bin_path.join("build"),
+    )
+    .map_err(|e| AssembleBuildpackDirectoryError::WriteBuildBinary(bin_path.join("build"), e))?;
+
+    create_file_symlink("build", bin_path.join("detect")).map_err(|e| {
+        AssembleBuildpackDirectoryError::WriteDetectBinary(bin_path.join("detect"), e)
+    })?;
+
+    if !buildpack_binaries.additional_target_binary_paths.is_empty() {
+        let additional_binaries_dir = destination_path
+            .join(".libcnb-cargo")
+            .join("additional-bin");
+
+        fs::create_dir_all(&additional_binaries_dir).map_err(|e| {
+            AssembleBuildpackDirectoryError::CreateAdditionalBinariesDirectory(
+                additional_binaries_dir.clone(),
+                e,
+            )
+        })?;
+
+        for (binary_target_name, binary_path) in &buildpack_binaries.additional_target_binary_paths
+        {
+            fs::copy(
+                binary_path,
+                additional_binaries_dir.join(binary_target_name),
+            )
+            .map_err(|e| {
+                AssembleBuildpackDirectoryError::WriteAdditionalBinary(
+                    additional_binaries_dir.join(binary_target_name),
+                    e,
+                )
+            })?;
+        }
+    }
+
+    Ok(())
+}
+
+/// Creates a meta-buildpack directory and copies all required meta-buildpack assets to it.
+///
+/// This function will not validate if the buildpack descriptor at the given path is valid and will
+/// use it as-is.
+///
+/// It will also rewrite all package.toml references that use the `libcnb:{buildpack_id}` format as
+/// well as relative file references to use absolute paths.
+///
+/// # Errors
+///
+/// Will return `Err` if the meta-buildpack directory could not be assembled.
+pub fn assemble_meta_buildpack_directory(
+    destination_path: impl AsRef<Path>,
+    buildpack_source_dir: impl AsRef<Path>,
+    buildpack_path: impl AsRef<Path>,
```

Comment:
* `buildpack_path` is the `buildpack.toml` file 
* `buildpack_source_dir` which is the directory that contains the `buildpack.toml` file.  
 
the `buildpack_source_dir` could be derived from `buildpack_path` though since it's the parent directory. alternatively, the `buildpack_path` could be replaced with `BuildpackData` which might add some clarity to the signature as well.